3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/PiiPseudonymizerEnrichment.scala

```diff
@@ -351,3 +479,69 @@ case class PiiStrategyPseudonymize(hashFunction: MessageDigest) extends PiiStrat
   override def scramble(clearText: String): String = hash(clearText)
   def hash(text: String): String                   = String.format("%064x", new java.math.BigInteger(1, hashFunction.digest(text.getBytes(TextEncoding))))
 }
+
+/**
+ * The mutator class encapsulates the mutator function and the field name where the mutator will be applied.
+ */
+case class Mutator(fieldName: String, muatatorFn: PiiConstants.MutatorFn)
+
+/**
+ * Case class for capturing scalar field modifications.
+ */
+case class ScalarModifiedField(fieldName: String, originalValue: String, modifiedValue: String) extends ModifedField
+
+/**
+ * Case class for capturing JSON field modifications.
+ */
+case class JsonModifiedField(field: String, originalValue: String, modifiedValue: String, jsonPath: String, schema: String)
+    extends ModifedField
+
+/**
+ * Parent class for classes that serialize the values that were modified during the PII enrichment.
+ */
+case class PiiModifiedFields(val modifiedFields: PiiConstants.ModifiedFields, val strategy: PiiStrategy)
```

Comment:
can we make all those final? I don't feel good about publicly-exposed non final case class